packages/server/postgres/utils/checkTeamEq.ts

```diff
@@ -0,0 +1,72 @@
+import Team from '../../database/types/Team'
+import getTeamsById, {IGetTeamsByIdResult} from '../../postgres/queries/getTeamsById'
+import getRethink from '../../database/rethinkDriver'
+import lodash from 'lodash'
+import {CustomResolver, checkTableEq} from './checkEqBase'
+
+const namesAreEqual: CustomResolver = (rethinkName, pgName) =>
+  lodash.isEqual(rethinkName, pgName) || rethinkName.slice(0, 100) === pgName
+
+const alwaysDefinedFieldsCustomResolvers = {
+  name: namesAreEqual,
+  createdAt: undefined,
+  isArchived: undefined,
+  isPaid: undefined,
+  tier: undefined,
+  orgId: undefined,
+  updatedAt: undefined
+} as {
+  [Property in keyof Partial<Team>]: CustomResolver | undefined
+}
+
+const maybeUndefinedFieldsDefaultValues = {
+  jiraDimensionFields: [],
+  lastMeetingType: 'retrospective'
+} as {
+  [Property in keyof Partial<Team>]: any
+}
+
+const getPairNeFields = (rethinkTeam: Team, pgTeam: IGetTeamsByIdResult): string[] => {
```

Comment:
yup! i was holding off on refactoring this part, but i think the API should be stable now. the only thing I changed in the generic function is the shape of `maybeUndefinedFields` now has tuple with `[<:CustomResolver>, <:defaultPgValue>]` as values, since a field can both maybe be undefined and maybe need a custom resolver.

https://github.com/ParabolInc/parabol/pull/4856/files#diff-9f822689a32c27c0e504373b650b2638acee971898aa27f930dec63e2f86b0d2R38-R42